/*
 * Copyright 2018- The Pixie Authors.
 * Modifications Copyright 2023- Gimlet Labs, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

syntax = "proto3";

package gml.uuidpb;

option go_package = "uuidpb";

message UUID {
  // Big ending style encoding similar to what a UUID Variant 1 would do.
  // For example, 00112233-4455-6677-8899-aabbccddeeff should get encoded as:
  // high_bits: 0x0011223344556677
  // low_bits:  0x8899aabbccddeeff
  uint64 high_bits = 2;
  uint64 low_bits = 3;
  reserved 1;
}
